src/core/Tools.cpp

```diff
@@ -116,9 +150,24 @@ namespace Tools
         return formatsStringList.join(" ");
     }
 
+    bool hexDecode(uchar* dest, size_t destSize, const char* src, size_t srcSize)
```

Comment:
I'm aware of [`QByteArray::fromHex()`](https://doc.qt.io/qt-5/qbytearray.html#fromHex). However, that necessarily results in dynamic memory allocation, which is not ideal for my use case, where I need to do this ~500 million times in a loop.